osmosis/src/storeProvider.js

```diff
@@ -1,4 +1,11 @@
 export const StoreProvider = (storeProviders, wrappedComponent) => {
-  storeProviders.reverse().forEach(provider => (wrappedComponent = provider(wrappedComponent)));
-  return wrappedComponent;
+  if(!Array.isArray(storeProviders)) throw new Error('StoreProvider requires an array of wrapper functions')
+  if(!wrappedComponent) throw new Error('StoreProvider requires the root component of your app')
+
+  try {
+    storeProviders.reverse().forEach(provider => (wrappedComponent = provider(wrappedComponent)));
+    return wrappedComponent;
+  } catch(error) {
+    throw new Error('something went wrong with StoreProvider arguments:\n' + error)
```

Comment:
@chaceburnette I wasn't sure what error message to put here. If we just throw the actual error itself without extracting its message, it would look likes this: 

<img width="446" alt="Screen Shot 2021-03-29 at 12 48 16" src="https://user-images.githubusercontent.com/25942541/112892280-f9c29700-908d-11eb-8968-eb6530153636.png">